homeassistant/components/music_assistant/media_player.py

```diff
@@ -0,0 +1,751 @@
+        self._attr_media_position_updated_at = None
+        self._attr_media_position = None
+        self._attr_media_duration = None
+        self._attr_media_album_artist = None
+        self._attr_media_artist = None
+        self._attr_media_album_name = None
+        self._attr_media_title = None
+        self._attr_media_content_id = None
+        self._attr_media_content_type = "music"
+        self._attr_media_image_url = None
+        self._prev_time: float = 0
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        await super().async_added_to_hass()
+        # we need to get the hass object in order to get our config entry
+        # and expose the player to the conversation component, assuming that
+        # the config entry has the option enabled.
+        await self._expose_players_assist()
+
+        # we subscribe to player queue time update but we only
+        # accept a state change on big time jumps (e.g. seeking)
+        async def queue_time_updated(event: MassEvent) -> None:
+            if event.object_id != self.player.active_source:
+                return
+            if abs((self._prev_time or 0) - event.data) > 5:
+                await self.async_on_update()
+                self.async_write_ha_state()
+            self._prev_time = event.data
+
+        self.async_on_remove(
+            self.mass.subscribe(
+                queue_time_updated,
+                EventType.QUEUE_TIME_UPDATED,
+            )
+        )
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any]:
+        """Return additional state attributes."""
+        player = self.player
+        if TYPE_CHECKING:
+            assert player.active_source is not None
+        queue = self.mass.player_queues.get(player.active_source)
+        attrs = {
+            ATTR_MASS_PLAYER_ID: self.player_id,
+            ATTR_MASS_PLAYER_TYPE: player.type.value,
+            ATTR_GROUP_LEADER: player.synced_to,
+            ATTR_ACTIVE_QUEUE: queue.queue_id if queue else None,
+            ATTR_ACTIVE_GROUP: player.active_group,
+            ATTR_QUEUE_ITEMS: queue.items if queue else None,
+            ATTR_QUEUE_INDEX: queue.current_index if queue else None,
+        }
+        # add optional stream_title for radio streams
+        if (
+            queue
+            and queue.current_item
+            and queue.current_item.streamdetails
+            and queue.current_item.streamdetails.stream_title
+        ):
+            attrs[ATTR_STREAM_TITLE] = queue.current_item.streamdetails.stream_title
+        elif (
+            queue
+            and queue.current_item
+            and queue.current_item.media_type == MediaType.RADIO
+            and queue.current_item.media_item
+            and queue.current_item.media_item.metadata
+        ):
+            attrs[ATTR_STREAM_TITLE] = (
+                queue.current_item.media_item.metadata.description
+            )
+        return attrs
```

Comment:
Okay so ideally we have as little state attributes as possible, as they are all stored in the database and that impacts performance. We much rather make data objects (like the queue) available via a service action

Reply:
hmmmm, I'm not sure I like that as these attributes are really important to sensefully automate a bit with MA players. Service actions are not user friendly at all. These few attributes wont hurt db performance at all.

Let's go through this list to determine which ones we really like to keep and which ones are too much of an edge case but dropping them all is not acceptable imo

Reply:
Do note that these are simple attributes. E.g. queue_items is just the count and not the actual queue items. THAT would be a separate action indeed.

Reply:
> These few attributes wont hurt db performance at all.

If we start using this argument with every state attribute someone wants to add we will run into the problem eventually. As in, we've denied a lot of extra state attributes recently, so I don't want to directly say yes. I agree that we need to have a good look at what is included and evaluate on a case by case basis

Reply:
The additional attributes are very much appreciated by (custom) card developers and people who just like to create a rich dashboard so please leave them as much as we can. This gives little to no overhead.

Reply:
Proposal:

Let's keep:
- mass player type
- active queue
- stream title

In a follow up PR: Add action (with return value) to get the active queue details.

Recognizing if its a mass player can be done because the presence of the "mass player type" - there is no real use for keeping mass player id (at least I wouldnt know what).

Reply:
We can also drop stream title if we parse the artist - title from it but then we need to accept the fact that some radio stations make a mess of this and swap artist and title here. Stream title is the string that is being sent by radio stations what is currently on air. It is freeform text so a bit of cowboy land

Reply:
How does Sonos do that? I play radio on there regularly, and I never really had any issues with how its displayed

Reply:
> How does Sonos do that? I play radio on there regularly, and I never really had any issues with how its displayed

They have deals with tune in to get the metadata. We just get the "ICY" metadata streamtitle, which is freeform.

Reply:
> Proposal:
> 
> Let's keep:
> 
> * mass player type
> * active queue
> * stream title
> 
> In a follow up PR: Add action (with return value) to get the active queue details.
> 
> Recognizing if its a mass player can be done because the presence of the "mass player type" - there is no real use for keeping mass player id (at least I wouldnt know what).

I removed the other attrs for now.